src/safeposix/syscalls/net_calls.rs

```diff
@@ -1,5 +1,1505 @@
+                                    *optval = 1;
+                                } else {
+                                    *optval = 0;
+                                }
+                            }
+                            //handling the ignored buffer settings:
+                            SO_SNDBUF => {
+                                *optval = sockfdobj.sndbuf;
+                            }
+                            SO_RCVBUF => {
+                                *optval = sockfdobj.rcvbuf;
+                            }
+                            //returning the type if asked
+                            SO_TYPE => {
+                                *optval = sockfdobj.socktype;
+                            }
+                            //should always be true
+                            SO_OOBINLINE => {
+                                *optval = 1;
+                            }
+                            SO_ERROR => {
+                                let tmp = sockfdobj.errno;
+                                sockfdobj.errno = 0;
+                                *optval = tmp;
+                            }
+                            _ => {
+                                return syscall_error(Errno::EOPNOTSUPP, "getsockopt", "unknown optname passed into syscall");
+                            }
+                        }
+                    }
+                    _ => {
+                        return syscall_error(Errno::EOPNOTSUPP, "getsockopt", "unknown level passed into syscall");
+                    }
+                }
+            } else {
+                return syscall_error(Errno::ENOTSOCK, "getsockopt", "the provided file descriptor is not a socket");
+            }
+        } else {
+            return syscall_error(Errno::EBADF, "getsockopt", "the provided file descriptor is invalid");
+        }
+        return 0;
+    }
+
+    //int setsockopt(int sockfd, int level, int optname, const void *optval, socklen_t optlen);
+    pub fn setsockopt_syscall(&self, fd: i32, level: i32, optname: i32, optval: i32) -> i32 {
+        let fdtable = self.filedescriptortable.read().unwrap();
+        
+        if let Some(wrappedfd) = fdtable.get(&fd) {
+            let mut filedesc = wrappedfd.write().unwrap();
+            if let Socket(sockfdobj) = &mut *filedesc {
+                //checking that we recieved SOL_SOCKET\
+                match level {
+                    SOL_UDP => {
+                        return syscall_error(Errno::EOPNOTSUPP, "getsockopt", "UDP is not supported for getsockopt");
+                    }
+                    SOL_TCP => {
+                        if optname == TCP_NODELAY {
+                            return 0;
+                        }
+                        return 0; //temp for Apache
+                        //return syscall_error(Errno::EOPNOTSUPP, "getsockopt", "TCP options not remembered by getsockopt");
```

Comment:
Which one of these should we go with?

Reply:
I don't know. I didn't write this one I think @TristanB22 did but this is probably copied from what we were doing in repy safeposix so maybe @youngfalcon has some idea.

Reply:
I did not work on this syscall, unfortunately. So I don't know either.

Reply:
![image](https://user-images.githubusercontent.com/67669475/150911688-14aff7e4-a924-464e-ab5c-3f9b608c3634.png)

Reply:
This was in Repy... I just ported it over to keep the same kind of functionality. Not sure why it was there beyond that.